packages/codecs/src/did.ts

```diff
@@ -0,0 +1,30 @@
+import { refinement, string } from 'codeco'
+
+import type { Opaque } from './types.js'
```

Comment:
Since some time `ts-essentials` package provides working `Opaque` type. We'd need that in other places, so IMO, it makes sense now to use `Opaque` from `ts-essentials` as dev dependency here.

Reply:
To be honest, the only reason for using custom `Opaque` was that the one from `ts-essentials` [did not work](https://github.com/ts-essentials/ts-essentials/pull/348)

Reply:
OK great that you got the upstream package fixed! I'll update the PR.
Shouldn't `ts-essentials` be a regular dependency rather than a dev one to ensure consuming packages also have it installed?

Reply:
`Opaque`, `NonEmptyArray` and most of the other goodies there are only type declarations. I conclude, it makes `ts-essentials` a dev dependency.